airflow/providers/databricks/provider.yaml

```diff
@@ -76,6 +76,7 @@ dependencies:
   - pandas>=2.1.2,<2.2;python_version>="3.9"
   - pandas>=1.5.3,<2.2;python_version<"3.9"
   - pyarrow>=14.0.1
+  - azure-identity>=1.3.1
```

Comment:
I assume not all databricks users use azure so this shouldn't be listed on `dependencies` it should be on `additional-extras` thus allowing each databricks user to decide if they want to install it.

Reply:
Agree

Reply:
Though if it is, it will not work without some modifications (moving the imports to local imports and handling import failure) it will not work for people who have no `azure-identity` installed. So this PR will have to be reworked slightly when azure-identity is put to "additional-extras" to handle case where it's not installed propely and skip the AZURE identity checks.

Reply:
Yep, same thought here

Reply:
Thank you all for the input here. I've changed the code to address the issues. Let me know your thoughts.